Source/EbsdLib/TSL/AngReader.cpp

```diff
@@ -98,54 +98,56 @@ AngReader::AngReader()
 // -----------------------------------------------------------------------------
 AngReader::~AngReader()
 {
-  deletePointers();
-}
-
-// -----------------------------------------------------------------------------
-//
-// -----------------------------------------------------------------------------
-void AngReader::initPointers(size_t numElements)
-{
-  setNumberOfElements(numElements);
-  size_t numBytes = numElements * sizeof(float);
-  m_Phi1 = allocateArray<float>(numElements);
-  m_Phi = allocateArray<float>(numElements);
-  m_Phi2 = allocateArray<float>(numElements);
-  m_Iq = allocateArray<float>(numElements);
-  m_Ci = allocateArray<float>(numElements);
-  m_PhaseData = allocateArray<int>(numElements);
-  m_X = allocateArray<float>(numElements);
-  m_Y = allocateArray<float>(numElements);
-  m_SEMSignal = allocateArray<float>(numElements);
-  m_Fit = allocateArray<float>(numElements);
-
-  ::memset(m_Phi1, 0, numBytes);
-  ::memset(m_Phi, 0, numBytes);
-  ::memset(m_Phi2, 0, numBytes);
-  ::memset(m_Iq, 0, numBytes);
-  ::memset(m_Ci, 0, numBytes);
-  ::memset(m_PhaseData, 0, numBytes);
-  ::memset(m_X, 0, numBytes);
-  ::memset(m_Y, 0, numBytes);
-  ::memset(m_SEMSignal, 0, numBytes);
-  ::memset(m_Fit, 0, numBytes);
-}
-
-// -----------------------------------------------------------------------------
-//
-// -----------------------------------------------------------------------------
-void AngReader::deletePointers()
-{
-  this->deallocateArrayData<float>(m_Phi1);
-  this->deallocateArrayData<float>(m_Phi);
-  this->deallocateArrayData<float>(m_Phi2);
-  this->deallocateArrayData<float>(m_Iq);
-  this->deallocateArrayData<float>(m_Ci);
-  this->deallocateArrayData<int>(m_PhaseData);
-  this->deallocateArrayData<float>(m_X);
-  this->deallocateArrayData<float>(m_Y);
-  this->deallocateArrayData<float>(m_SEMSignal);
-  this->deallocateArrayData<float>(m_Fit);
+  if(m_Phi1Cleanup)
+  {
+    deallocateArrayData<float>(m_Phi1);
+    m_Phi1 = nullptr;
+  }
+  if(m_PhiCleanup)
+  {
+    deallocateArrayData<float>(m_Phi);
+    m_Phi = nullptr;
+  }
+  if(m_Phi1Cleanup)
+  {
+    deallocateArrayData<float>(m_Phi1);
+    m_Phi1 = nullptr;
```

Comment:
`if(m_Phi1Cleanup) { ... }`  exists twice, once before `if(m_PhiCleanup)` and once after.  One of these should be removed.

Reply:
Good catch. Should have been phi2 for the second one.